docs: Update docs framework

ronny-mysten opened the pull request:
## Description

Updates the documentation framework from mdbook to docusaurus. Content remains largely untouched. Content overhaul to be in a separate PR.

## Test plan

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.
For each box you select, include information after the relevant heading that describes the impact of your changes that
a user might notice and any actions they must take to implement updates. (Add release notes after the colon for each item)

- [ ] Storage node:
- [ ] Aggregator:
- [ ] Publisher:
- [ ] CLI:

wbbradley commented on .github/actions/set-up-walrus/action.yaml:
```diff
@@ -65,8 +65,10 @@ runs:
         INPUT_SUI_NETWORK: ${{ inputs.SUI_NETWORK }}
         INPUT_WALRUS_CONFIG: ${{ inputs.WALRUS_CONFIG }}
       run: |
+        curl https://storage.googleapis.com/mysten-walrus-binaries/walrus-${{ inputs.SUI_NETWORK }}-latest-ubuntu-x86_64 -o bin/walrus
```
This line appears to be redundant with the next line, but slightly different. Maybe a typo?

wbbradley replied:
What is the change here attempting to address?

wbbradley commented on documentation/blog/authors.bak:
```diff
@@ -0,0 +1,25 @@
+yangshun:
```
Inclusion of this file looks unintentional.

ronny-mysten commented on .editorconfig:
```diff
@@ -18,6 +18,11 @@ trim_trailing_whitespace = true
+# MDX files should follow the same rules as markdown but allow longer lines
+[*.mdx]
+indent_size = unset
+max_line_length = unset
```
But how is this a good editing experience?
<img width="857" height="373" alt="image" src="https://github.com/user-attachments/assets/cf413925-1a67-40bb-8c76-a08320874871" />

ronny-mysten replied:
My bad. Should have looked closer. The current docs content has no indent, line length, or whitespace rules, so just reusing the current setting.

ronny-mysten commented:
Took docs/site out of the editor ignore list. 
Configured the mdx content to match the current docs/*.md rules exactly
Added react files to the current list of frontend rules
Added the lockfile and generated files to the ignore list (docs/site/src/.generated/*,docs/site/static/glossary.json)

wbbradley reviewed (approved):
Thanks for working on this!
